src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -969,28 +999,27 @@ private StringBuffer preMigrationChecks(StringBuffer itemStatus,
 					if (webLinkUrl == null || webLinkUrl.isEmpty())
 					{
 						status.append("Link "+ fileName + " could not be migrated due to empty URL link. ");
-						return new AsyncResult<String>(status.toString());
+						return new AsyncResult<String>(setUploadJobEndtimeStatus(id, status));
 					}
 					try {
 						// special handling of Web Links resources
 						content = new ByteArrayInputStream(Utils.getWebLinkContent(
 								fileName, webLinkUrl).getBytes());
-					} catch (java.net.MalformedURLException e) {
+					} catch (Exception e) {
 						// return status with error message
 						status.append("Link "
 								+ fileName
 								+ " could not be migrated. Please change the link name to be the complete URL and migrate the site again.");
-						return new AsyncResult<String>(status.toString());
+						return new AsyncResult<String>(setUploadJobEndtimeStatus(id, status));
 					}
 				}
-			} catch (java.io.IOException e) {
+			} catch (Exception e) {
```

Comment:
If we catch the unspecific "Exception" the exception really needs to be logged since we don't know what actually happened, and we can't fix production if we don't know what the problem is.  This applies to every where we catch the generic "Exception".